### PR TITLE
Implement Restful

### PR DESCRIPTION
**Pull request by hoangphanusc**
Added Restful model to the program. Please check the commit message for the detailed description of changes.

### CODE REVIEW

File: usecases/usecases.go [hunk 1]
```diff
@@ -236,6 +237,15 @@ func (interactor *ProfileInteractor) AddGame(userId, libraryId int, gameName, ga
...
 		return 0, err, 403
 	}
+	existed, err := interactor.GameRepository.GameExisted(gameName, libraryId)
+	if err != nil {
+		return 0, err, 500
+	}
+	if existed {
+		err := fmt.Errorf("Game '%s' is already in library", gameName)
+		// interactor.Logger.Log(err.Error())
+		return 0, err, 403
```
Comment (trietphm) [inline usecases/usecases.go:247]: `403`? User has permission to do this. It should be `400 - invalid request`

Comment (hoangphanusc) [inline usecases/usecases.go:247]: Makes sense. The same for `403` "Username [...] is taken"?

Comment (trietphm) [inline usecases/usecases.go:247]: Right

**Comment by hoangphanusc** [comment]
And am I good to merge?

**Comment by trietphm** [comment]
Yup, it's ok.

**Pull request merged**